pkg/controller/common/reconciler/reconciler.go

```diff
@@ -88,8 +88,21 @@ func ReconcileResource(params Params) error {
 	namespace := params.Expected.GetNamespace()
 	name := params.Expected.GetName()
 	log := ulog.FromContext(params.Context).WithValues("kind", kind, "namespace", namespace, "name", name)
+	deleted := false
 
 	create := func() error {
+		if deleted {
+			// check if not being deleted
+			err = params.Client.Get(params.Context, types.NamespacedName{Name: name, Namespace: namespace}, params.Reconciled)
+			if err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+			if !params.Reconciled.GetDeletionTimestamp().IsZero() {
+				log.Info("Waiting for resource to be created because the old one is being deleted")
```

Comment:
"Waiting" seems a bit misleading to me. With this change it is up to the caller to detect that the resource has actually not been reconciled, and try again later. By "swallowing"/"hiding" the conflict and returning `nil` here we are assuming that the caller is going to do another attempt, but we can't be sure of that? This makes me feel that it should be up to the caller to decide what to do/log in case of a conflict, by using `apierrors.IsAlreadyExists()`

Reply:
You are right, we can't be 100% sure that the caller will retry. You proposal looks a good idea.

This means we will delete and (try) to create until the deletion is effective.

The only thing that tickles me is that with the current log messages it can be a little confusing.

From LoadBalancer to ClusterIP:
```scala
Deleting resource as it cannot be updated, it will be recreated
Deleted resource successfully
Creating resource
Deleting resource as it cannot be updated, it will be recreated
Deleted resource successfully
Creating resource
Deleting resource as it cannot be updated, it will be recreated
Deleted resource successfully
Creating resource
Deleting resource as it cannot be updated, it will be recreated
Deleted resource successfully
Creating resource
Deleting resource as it cannot be updated, it will be recreated
Deleted resource successfully
Creating resource
Deleting resource as it cannot be updated, it will be recreated
Deleted resource successfully
Creating resource
Deleting resource as it cannot be updated, it will be recreated
Deleted resource successfully
Creating resource
Creating resource
Created resource successfully
Ensuring no voting exclusions are set
Ensuring no voting exclusions are set
Ensuring no voting exclusions are set
```

Reply:
> The only thing that tickles me is that with the current log messages it can be a little confusing.

Is it when we requeue immediately, or when we just ignore the error in the driver? Maybe we could add an additional log in that specific case.

Reply:
It is when we ignore the error and then requeue in the driver (at the caller level), see 0218fd6db.